### ISSUE TITLE
Give house number quest a higher priority

### ISSUE THREAD

**Issue by schmittlauch** [streetcomplete/StreetComplete#2473]
**Use case**
In my opinion, adding missing house numbers and addresses is one of the most important tasks as it massively improves the user experience of people accessing the OSM – maybe even for the first time – through some search or geocoder.
While I understand the intention of asking for house numbers only after the building type has been determined, asking for the address/ house number shall be the next question for building types that can have one.
Instead the next questions I'm being asked are about the number of levels and the roof shape. Only after answering these or disabling these quest types I finally get the chance to enter a house number.

**Proposed Solution**

The house number quest needs to be prioritised such that is the first one to appear for houses with known types. Asking for number of house levels/ storeys and roof shape need to get lower priority and come after that.

**Comment by westnordost** [comment]
Please search the issue tracker before you open a new ticket: #2464

**Comment by matkoniecz** [comment]
> Asking for number of house levels/ storeys and roof shape need to get lower priority and come after that.

It has lower priority, asking about addresses happens later anyway for technical reasons. Filtering buildings for address quest is quite complex, while filtering for building levels and roof shape is simpler.

**Comment by schmittlauch** [comment]
While #2464 demands that house numbers shall be editable *before* adding a building type, I accept the design decision and rationale behind having to add building types first but think house numbers are more valuable information than roof types and shall be asked immediately afterwards.

My use case: I was walking through some villages in the middle of nowhere, not coming back there again, and could not stop for a minute at each building to add all the possible information. My guess is that adding house type and number is more valuable than number of levels in this limited time.

The higher filtering complexity is unfortunate, so if the higher information value isn't worth the higher filtering complexity that's the way it is then.

**Comment by matkoniecz** [comment]
> and shall be asked immediately afterwards

This is happening this way due to technical limitation, not due to deliberate decision to make house numbers lower priority. 

See https://github.com/streetcomplete/StreetComplete/issues/2438#issuecomment-754031120

I would estimate that it would take about 450 hours of work for me to implement it (including time for design, learning, testing, refactoring, bugfixing).
